File: python3/binary_gap.py
# Space : O(1)
# Time  : O(bin len)

class Solution:
    def binaryGap(self, n: int) -> int:
        ans = 0
        b = bin(n)[2:]

        s = 0
        b_len = len(b)
        for i in range(1, b_len):
            if b[i] == '1':
                ans = max(ans, i-s)
                s = i

        return ans

File: golang/binary_gap.go
package main

import "fmt"

// Space : O(1)
// Time  : O(bin len)

func binaryGap(n int) int {
    b := fmt.Sprintf("%b", n)
    b_len := len(b)
    s, ans := 0, 0
    
    for i := 1; i<b_len; i++ {
        if string(b[i]) == "1" {
            ans = Max(ans, i-s)
            s = i
        }
    }
    
    return ans
}